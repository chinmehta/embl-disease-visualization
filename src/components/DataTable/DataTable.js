import React from "react";
import TableRow from "../TableRow/TableRow";
import TableCellHeader from "../TableCellHeader/TableCellHeader";

function DataTable({ tableDataHeader, tableData }) {
  return (
    <table className="sm:w-full lg:w-4/5 table-auto border-solid border-gray-400 border-collapse mb-10">
      <thead>
        <tr>
          {tableDataHeader &&
            tableDataHeader.map((headerElement, index) => (
              <TableCellHeader key={index} text={headerElement} />
            ))}
        </tr>
      </thead>
      <tbody className="border border-gray-400 border-collapse">
        {tableData &&
          tableData.map((dataRow) => (
            <TableRow key={dataRow.id} data={dataRow} />
          ))}
      </tbody>
    </table>
  );
}

export default DataTable;
