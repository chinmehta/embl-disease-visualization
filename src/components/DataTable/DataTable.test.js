import React from "react";
import renderer from "react-test-renderer";
import DataTable from "./DataTable";

it("renders when no data  is passed as prop", () => {
  const element = renderer.create(<DataTable />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when data  is passed as prop", () => {
  const tableData = [
    {
      approvedSymbol: "EGFR",
      approvedName: "epidermal growth factor receptor",
      score: 0.890384,
      label: ["label1", "label2"],
      chartDataArray: [0.4, 0.9],
    },
    {
      approvedSymbol: "KRAS",
      approvedName: "KRAS proto-oncogene, GTPase",
      score: 0.790384,
      label: ["label1", "label2"],
      chartDataArray: [0.4, 0.9],
    },
  ];
  const tableDataHeader = [
    "",
    "Approved Symbol",
    "Gene Name",
    "Overall Association Score",
  ];
  const element = renderer
    .create(
      <DataTable tableData={tableData} tableDataHeader={tableDataHeader} />
    )
    .toJSON();

  expect(element).toMatchSnapshot();
});
