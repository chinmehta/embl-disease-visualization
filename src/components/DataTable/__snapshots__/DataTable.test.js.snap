// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when data  is passed as prop 1`] = `
<table
  className="sm:w-full lg:w-4/5 table-auto border-solid border-gray-400 border-collapse mb-10"
>
  <thead>
    <tr>
      <th
        className="font-bold py-4 px-2 leading-none border border-gray-400"
      >
        
      </th>
      <th
        className="font-bold py-4 px-2 leading-none border border-gray-400"
      >
        Approved Symbol
      </th>
      <th
        className="font-bold py-4 px-2 leading-none border border-gray-400"
      >
        Gene Name
      </th>
      <th
        className="font-bold py-4 px-2 leading-none border border-gray-400"
      >
        Overall Association Score
      </th>
    </tr>
  </thead>
  <tbody
    className="border border-gray-400 border-collapse"
  >
    <tr>
      <td
        className="flex items-stretch border border-gray-400 border-collapse p-0"
      >
        <button
          className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
          onClick={[Function]}
        >
          +
        </button>
      </td>
      <td
        className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
      >
        <a
          href="https://platform.opentargets.org/target/epidermal growth factor receptor"
          rel="noopener noreferrer"
          target="_blank"
        >
          EGFR
        </a>
      </td>
      <td
        className="px-2 border-collapse border border-gray-400"
      >
        epidermal growth factor receptor
      </td>
      <td
        className="px-2 border-collapse border border-gray-400"
      >
        0.890
      </td>
    </tr>
    <tr
      className="hidden"
    >
      <td
        className="border border-gray-400"
        colSpan="4"
      >
        <div
          className="tab-container my-2 mx-5"
        >
          <div
            className="tab-list flex border-gray-400 border-b-2"
          >
            <span
              className=" px-4 cursor-pointer bg-embl-primary-blue text-white"
              onClick={[Function]}
            >
              Bar chart
            </span>
            <span
              className=" px-4 cursor-pointer bg-white text-embl-primary-blue"
              onClick={[Function]}
            >
              Radar chart
            </span>
          </div>
          <div
            className="flex h-80 justify-center items-center"
          >
            <canvas
              className="max-h-full max-w-full"
              height={150}
              role="img"
              width={300}
            />
          </div>
        </div>
      </td>
    </tr>
    <tr>
      <td
        className="flex items-stretch border border-gray-400 border-collapse p-0"
      >
        <button
          className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
          onClick={[Function]}
        >
          +
        </button>
      </td>
      <td
        className="text-embl-primary-blue px-2 border-collapse border border-gray-400 leading-3"
      >
        <a
          href="https://platform.opentargets.org/target/KRAS proto-oncogene, GTPase"
          rel="noopener noreferrer"
          target="_blank"
        >
          KRAS
        </a>
      </td>
      <td
        className="px-2 border-collapse border border-gray-400"
      >
        KRAS proto-oncogene, GTPase
      </td>
      <td
        className="px-2 border-collapse border border-gray-400"
      >
        0.790
      </td>
    </tr>
    <tr
      className="hidden"
    >
      <td
        className="border border-gray-400"
        colSpan="4"
      >
        <div
          className="tab-container my-2 mx-5"
        >
          <div
            className="tab-list flex border-gray-400 border-b-2"
          >
            <span
              className=" px-4 cursor-pointer bg-embl-primary-blue text-white"
              onClick={[Function]}
            >
              Bar chart
            </span>
            <span
              className=" px-4 cursor-pointer bg-white text-embl-primary-blue"
              onClick={[Function]}
            >
              Radar chart
            </span>
          </div>
          <div
            className="flex h-80 justify-center items-center"
          >
            <canvas
              className="max-h-full max-w-full"
              height={150}
              role="img"
              width={300}
            />
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders when no data  is passed as prop 1`] = `
<table
  className="sm:w-full lg:w-4/5 table-auto border-solid border-gray-400 border-collapse mb-10"
>
  <thead>
    <tr />
  </thead>
  <tbody
    className="border border-gray-400 border-collapse"
  />
</table>
`;
